packages/dataorc-utils/src/dataorc_utils/config/defaults.py

```diff
@@ -0,0 +1,97 @@
+"""
+Environment configuration builders.
+
+Functions for building complete environment configurations from templates.
+"""
+
+from .enums import CoreParam
+
+
+def _generate_infra_names(config: dict, env_name: str, env_config: dict) -> None:
+    """Populate infrastructure-related names into `config` in-place.
+
+    Expects `env_config` to contain `env_suffix` and `location` when infra names
+    should be generated.
+    """
+    if "env_suffix" not in env_config or "location" not in env_config:
+        return
+
+    env_suffix = env_config["env_suffix"]
+    location = env_config["location"]
+
+    if "datalake_prefix" in env_config:
+        config[CoreParam.DATALAKE_NAME] = (
+            f"{env_config['datalake_prefix']}{env_suffix}{location}"
+        )
+
+    if "keyvault_prefix" in env_config:
+        config[CoreParam.AZ_KEYVAULT_SCOPE] = (
+            f"{env_config['keyvault_prefix']}-{env_name}-{location}"
+        )
+
+
+def _apply_env_overrides(config: dict, env_config: dict) -> None:
+    """Apply any remaining environment-specific overrides into `config` in-place."""
+    reserved = {
+        "env_suffix",
+        "location",
+        "keyvault_prefix",
+        "storage_prefix",
+        "datalake_prefix",
+    }
+    env_overrides = {k: v for k, v in env_config.items() if k not in reserved}
+    config.update(env_overrides)
+
+
+def build_environment_config(
+    env_name: str, env_config: dict, repository_defaults: dict | None = None
```

Comment:
Ref. the quick discussion we had at the office:

It seems like `repository_defaults` doesn't provide any specific functionality at the moment. Could it be removed?